Sources/MapboxSpeech/MapboxSpeech.swift

```diff
@@ -173,17 +173,18 @@ open class SpeechSynthesizer {
         request.setValue(userAgent, forHTTPHeaderField: "User-Agent")
         let task = URLSession.shared.dataTask(with: request as URLRequest) { (possibleData, possibleResponse, possibleError) in
             if let error = possibleError {
-                errorHandler(.unknown(response: possibleResponse, underlying: error, code: nil, message: nil))
```

Comment:
Would it make sense to also weakly hold reference to `self` to prevent any leaks in the future?

Reply:
Thanks for the suggestion.
I noticed that I could move switching to the main thread up to the `audioData(with:completionHandler:)`, so now I don't need to extract any private methods and use `self`.